### PR TITLE
(8498 ): Created vpc_creation resource

### PR DESCRIPTION
**Pull request by zhiqiangjin**
1. Created vpc_creation resource
2. Created acceptance test for vpc_creation resource
3. It supports create/refresh/destroy/import
4. It works in 4.3+ release

### CODE REVIEW

File: aviatrix/provider.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ func Provider() terraform.ResourceProvider {
 			"aviatrix_tunnel":                  resourceTunnel(),
 			"aviatrix_version":                 resourceAviatrixVersion(),
 			"aviatrix_vgw_conn":                resourceAviatrixVGWConn(),
+			"aviatrix_vpc_creation":            resourceVpcCreation(),
```
Comment (arvindsree) [inline aviatrix/provider.go]: Call it aviatrix_aws_vpc

File: aviatrix/resource_vpc_creation.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
+package aviatrix
+
+import (
+	"fmt"
+	"github.com/AviatrixSystems/go-aviatrix/goaviatrix"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+)
+
+func resourceVpcCreation() *schema.Resource {
```
Comment (arvindsree) [inline aviatrix/resource_vpc_creation.go]: rename to resourceAWSVPC

File: aviatrix/resource_vpc_creation.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+		Schema: map[string]*schema.Schema{
+			"cloud_type": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Type of cloud service provider.",
+			},
+			"account_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Account name. This account will be used to launch Aviatrix gateway.",
```
Comment (arvindsree) [inline aviatrix/resource_vpc_creation.go]: create an Aviatrix VPC

File: aviatrix/resource_vpc_creation.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+			},
+			"vpc_region": {
```
Comment (arvindsree) [inline aviatrix/resource_vpc_creation.go]: Just say region

Comment (arvindsree) [reply aviatrix/resource_vpc_creation.go]: Same for name and cidr

File: aviatrix/resource_vpc_creation.go [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+				Default:     false,
+				Description: "Specify the VPC as Aviatrix Transit VPC or not.",
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the VPC to be created.",
+			},
+			"subnets": {
+				Type:        schema.TypeList,
```
Comment (arvindsree) [inline aviatrix/resource_vpc_creation.go]: If it is just a list, would we not possible to automate for transit DMZ

File: aviatrix/resource_vpc_creation.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+			"aviatrix_transit_vpc": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specify the VPC as Aviatrix Transit VPC or not.",
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the VPC to be created.",
```
Comment (arvindsree) [inline aviatrix/resource_vpc_creation.go]: "that is " created since this is an output variable

File: aviatrix/resource_vpc.go [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+	vpcName := d.Get("vpc_name").(string)
+	if vpcName == "" {
+		id := d.Id()
+		log.Printf("[DEBUG] Looks like an import, no vpc names received. Import Id is %s", id)
+		d.Set("vpc_name", id)
+		d.SetId(id)
+	}
+
+	log.Printf("zjin00: nothing is wrong here")
```
Comment (arvindsree) [inline aviatrix/resource_vpc.go]: Remove debug prints

File: aviatrix/resource_vpc.go [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+	}
+
+	if vpc.AviatrixTransitVpc == "yes" {
+		log.Printf("[INFO] Creating a new Aviatrix Transit VPC: %#v", vpc)
+	} else {
+		log.Printf("[INFO] Creating a new VPC: %#v", vpc)
+	}
+
+	err := client.CreateVpc(vpc)
+	if err != nil {
```
Comment (arvindsree) [inline aviatrix/resource_vpc.go]: Should you ignore error if same VPC name is already created?

File: aviatrix/resource_vpc.go [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+	}
+
+	log.Printf("zjin00: nothing is wrong here")
+
+	vpc := &goaviatrix.Vpc{
+		VpcName: d.Get("vpc_name").(string),
+	}
+	vC, err := client.GetVpc(vpc)
+
+	log.Printf("zjin00: nothing is wrong here")
```
Comment (arvindsree) [inline aviatrix/resource_vpc.go]: Same

File: aviatrix/resource_vpc.go [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+
+	d.Set("cloud_type", vC.CloudType)
+	d.Set("account_name", vC.AccountName)
+	d.Set("vpc_region", vC.VpcRegion)
+	d.Set("vpc_name", vC.VpcName)
+	d.Set("vpc_cidr", vC.VpcCidr)
+	d.Set("aviatrix_transit_vpc", vC.AviatrixTransitVpc)
+	d.Set("vpc_id", vC.VpcID[0])
+
+	log.Printf("zjin00: nothing is wrong here")
```
Comment (arvindsree) [inline aviatrix/resource_vpc.go]: Remove

File: aviatrix/resource_vpc_test.go [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+    aws_access_key = "%s"
+    aws_secret_key = "%s"
+}
+
+resource "aviatrix_vpc" "test_vpc" {
+	cloud_type = 1
+	account_name = "${aviatrix_account.test_account.account_name}"
+	vpc_name = "tfg-%s"
+	vpc_region = "%s"
+	vpc_cidr = "%s"
```
Comment (arvindsree) [inline aviatrix/resource_vpc_test.go]: Hardcode the CIDR. Minimize the external variables that need to be passed

File: aviatrix/resource_vpc_creation.go (hunk 5 above)
Comment (zhiqiangjin) [reply aviatrix/resource_vpc_creation.go]: Done

File: aviatrix/resource_vpc.go (hunk 7 above)
Comment (zhiqiangjin) [reply aviatrix/resource_vpc.go]: Done

File: aviatrix/resource_vpc.go (hunk 10 above)
Comment (zhiqiangjin) [reply aviatrix/resource_vpc.go]: Done.

File: aviatrix/resource_vpc_creation.go (hunk 5 above)
Comment (zhiqiangjin) [reply aviatrix/resource_vpc_creation.go]: Done

File: aviatrix/resource_vpc.go (hunk 8 above)
Comment (zhiqiangjin) [reply aviatrix/resource_vpc.go]: Yes. Done.

File: aviatrix/resource_vpc.go (hunk 10 above)
Comment (zhiqiangjin) [reply aviatrix/resource_vpc.go]: Done.

**Pull request merged**